### PR TITLE
feat(explore): Implement metrics and columns popovers empty states

### PR DESCRIPTION
**Pull request by kgabryje**
### SUMMARY
This PR implements chart empty states for 4 scenarios:

1. No saved metrics
2. No saved expressions
3. No saved temporal expressions
4. No temporal columns

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
image

image

image

image

### TESTING INSTRUCTIONS
1. Create a chart using dataset without saved metrics, saved expressions and temporal columns
2. Check empty states in metrics popover, columns popover and temporal column popover

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

CC @kasiazjc

https://app.shortcut.com/preset/story/36403/explore-empty-states

### CODE REVIEW

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/ColumnSelectPopover.tsx [hunk 1]
```diff
@@ -234,48 +240,80 @@ const ColumnSelectPopover = ({
...
+                  key: calculatedColumn.column_name,
+                }))}
+              />
+            </FormItem>
+          ) : (
+            <EmptyStateSmall
+              image="empty.svg"
+              title={
+                isTemporal
+                  ? t('No time columns found')
```
Comment (villebro) [inline superset-frontend/src/explore/components/controls/DndColumnSelectControl/ColumnSelectPopover.tsx:273]: nit: time column sounds like it refers to `TIME` (hh:mm:ss) as opposed to DATE/DATETIME/TIMESTAMP.
```suggestion
                  ? t('No temporal columns found')
```

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/ColumnSelectPopover.tsx [hunk 2]
```diff
@@ -234,48 +240,80 @@ const ColumnSelectPopover = ({
...
-                label: simpleColumn.verbose_name || simpleColumn.column_name,
-                customLabel: (
-                  <StyledColumnOption column={simpleColumn} showType />
-                ),
-                key: simpleColumn.column_name,
-              }))}
+          {isTemporal && simpleColumns.length === 0 ? (
+            <EmptyStateSmall
+              image="empty.svg"
+              title={t('No time columns found')}
```
Comment (villebro) [inline superset-frontend/src/explore/components/controls/DndColumnSelectControl/ColumnSelectPopover.tsx:292]: Same here

**Review by villebro — COMMENTED** [review]
Two minor comments, other than that LGTM

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/ColumnSelectPopover.tsx (hunk 1 above)
Comment (kgabryje) [reply superset-frontend/src/explore/components/controls/DndColumnSelectControl/ColumnSelectPopover.tsx]: 👍

File: superset-frontend/src/explore/components/controls/DndColumnSelectControl/ColumnSelectPopover.tsx (hunk 2 above)
Comment (kgabryje) [reply superset-frontend/src/explore/components/controls/DndColumnSelectControl/ColumnSelectPopover.tsx]: Done

**Review by villebro — APPROVED** [review]
LGTM!

**Pull request merged**